src/components/add-products/add-products.js

```diff
@@ -1,38 +1,58 @@
 import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
 
+import { productActions } from '../../store/products/action';
 import BuildForm from '../form-builder';
 import ProductCard from './product-card';
-import { productActions } from '../../store/products/action';
 
 import './product.scss';
 
 function AddProducts() {
+  const id = Number(useParams().id);
+  const products = useSelector(({ products }) => products.products);
+  let product = null;
+  let completedForm = false;
   const dispatch = useDispatch();
   const [submitted, setSubmitted] = useState(false);
+
+  if (id !== 0 && products) {
+    product = products.find((object) => object.id === id);
+  }
+
   const [inputs, setInputs] = useState({
-    title: '',
-    price: '',
-    categoryId: 'Bread',
-    imageUrl: '',
+    title: product ? product.title : '',
+    price: product ? product.price : '',
+    categoryId: product ? product.categoryId : 'Bread',
+    imageUrl: product ? product.imageUrl : '',
   });
 
+  const { title, price, categoryId, imageUrl } = inputs;
   const onChange = (event) => {
     const { name, value } = event.target;
     setInputs((inputs) => ({ ...inputs, [name]: value }));
   };
 
+  if (title && price && categoryId && imageUrl) {
```

Comment:
Does this actually run on every update? Seems like it would fit best in the onChange handler.

Reply:
I don't believe I can put it in onChange, but I can (and did) put inside onSubmit so it only runs when the form goes to get submitted